genesis/engine/solvers/rigid/collider_decomp.py

```diff
@@ -171,25 +208,20 @@ def _init_collision_fields(self) -> None:
         )
 
         ############## broad phase SAP ##############
-        # This buffer stores the AABBs along the search axis of all geoms
-        struct_sort_buffer = ti.types.struct(value=gs.ti_float, i_g=gs.ti_int, is_max=gs.ti_int)
-        self.sort_buffer = struct_sort_buffer.field(
-            shape=self._solver._batch_shape(2 * self._solver.n_geoms_),
-            layout=ti.Layout.SOA,
+        self.broad_buffer = array_class.ColliderBroadPhaseBuffer(
```

Comment:
can you call it something like 
```
self.collision_global_info = CollisionGlobalInfo
```
and also put narrow-phase-related data into this wrapper? That way, all collision-related data are defined within this class.